pandas/tests/test_strings.py

```diff
@@ -2568,6 +2568,30 @@ def test_get(self):
         expected = Series(['3', '8', np.nan])
         tm.assert_series_equal(result, expected)
 
+    def test_get_complex(self):
+        # GH 20671, getting value not in dict raising `KeyError`
+        values = Series([(1, 2, 3), [1, 2, 3], {1, 2, 3},
+                         {1: 'a', 2: 'b', 3: 'c'}])
+
+        result = values.str.get(1)
+        expected = Series([2, 2, np.nan, 'a'])
+        tm.assert_series_equal(result, expected)
+
+        result = values.str.get(-1)
+        expected = Series([3, 3, np.nan, np.nan])
+        tm.assert_series_equal(result, expected)
+
+        for to_type in tuple, list, np.array:
```

Comment:
can you parameterize this

Reply:
not sure what do you mean by parametrize, sorry

Reply:
http://pandas-docs.github.io/pandas-docs-travis/contributing.html#using-pytest

something like

```
pytest.mark.parametrize("box", [tuple, list, np.array]):
def test.....(box):
   ...
```

Reply:
Didn't used that before, thanks for the clarification. Updated.